vcd/resource_vcd_vapp_vm.go

```diff
@@ -1390,17 +1417,35 @@ func resourceVmHotUpdate(d *schema.ResourceData, meta interface{}, vmType typeOf
 		return diag.FromErr(err)
 	}
 
-	if d.HasChange("sizing_policy_id") {
-		var sizingPolicy *types.VdcComputePolicy
-		value := d.Get("sizing_policy_id")
-		vdcComputePolicy, err := vcdClient.Client.GetVdcComputePolicyById(value.(string))
-		if err != nil {
-			return diag.Errorf("error getting sizing policy %s: %s", value.(string), err)
+	sizingId, newSizingId := d.GetChange("sizing_policy_id")
+	placementId, newPlacementId := d.GetChange("placement_policy_id")
+
+	sizingPolicyChanged := d.HasChange("sizing_policy_id")
+	placementPolicyChanged := d.HasChange("placement_policy_id")
+	if !sizingPolicyChanged {
+		// As sizing_policy_id is Computed+Optional, the only way to unset it should be to write `sizing_policy_id = ""`
+		// in the HCL. However, when this is done, Terraform SDK doesn't detect this change: d.HasChange() returns false,
+		// d.GetChange returns both old values and d.Get returns old value, hence `sizingPolicyChanged` will be always false.
+		// We need to inspect the raw HCL to get the correct value.
```

Comment:
Can you possibly now that this is the case with `terraform-plugin-sdk v2.24.0`. This can change in future and should make it easier to look for new/improved solutions

Reply:
Done!